src/components/support-form/SupportForm.tsx

```diff
@@ -0,0 +1,374 @@
+      volunteer: yup.bool(),
+      associationMember: yup.bool(),
+      promoter: yup.bool(),
+    })
+    .required()
+    .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked),
+  benefactor: yup.object().when('roles.benefactor', {
+    is: true,
+    then: yup
+      .object()
+      .shape({
+        campaignBenefactor: yup.bool(),
+        platformBenefactor: yup.bool(),
+      })
+      .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked),
+  }),
+  associationMember: yup.object().when('roles.associationMember', {
+    is: true,
+    then: yup
+      .object()
+      .shape({
+        isMember: yup.bool(),
+      })
+      .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked),
+  }),
+  partner: yup.object().when('roles.partner', {
+    is: true,
+    then: yup
+      .object()
+      .shape({
+        npo: yup.bool(),
+        bussiness: yup.bool(),
+        other: yup.bool(),
+        otherText: yup.string().when('partner.other', {
+          is: true,
+          then: yup.string(),
+        }),
+      })
+      .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked)
+      .test('CustomValidation', 'Custom validation', function (this: any, values: any) {
+        const { path, createError } = this
+        return values.other && !values.otherText
+          ? createError({ path, message: 'field is required' })
+          : true
+      }),
+  }),
+  volunteer: yup.object().when('roles.volunteer', {
+    is: true,
+    then: yup.object().shape({
+      areas: yup.array().of(yup.string()),
+    }),
+  }),
+  promoter: yup.object().when('roles.promoter', {
+    is: true,
+    then: yup
+      .object()
+      .shape({
+        mediaPartner: yup.bool(),
+        ambassador: yup.bool(),
+        other: yup.bool(),
+        otherText: yup.string(),
+      })
+      .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked)
+      .test('CustomValidation', 'Custom validation', function (this: any, values: any) {
+        const { path, createError } = this
+        return values.other && !values.otherText
+          ? createError({ path, message: 'field is required' })
+          : true
+      }),
+  }),
+})
+
+const defaults: SupportFormData = {
+  terms: false,
+  info: {
+    email: '',
+    name: '',
+    phone: '',
+    address: '',
+  },
+  roles: {
+    benefactor: false,
+    partner: false,
+    associationMember: false,
+    promoter: false,
+    volunteer: false,
+  },
+  benefactor: {
+    campaignBenefactor: false,
+    platformBenefactor: false,
+  },
+  partner: {
+    npo: false,
+    bussiness: false,
+    other: false,
+    otherText: '',
+  },
+  volunteer: {
+    areas: [],
+  },
+  associationMember: {
+    isMember: true,
+  },
+  promoter: {
+    mediaPartner: false,
+    ambassador: false,
+    other: false,
+    otherText: '',
+  },
+}
+
+export default function Steppers(this: any) {
```

Comment:
It's better to match the default export of the component with the filename `SupportForm`

This enhances the IDE suggestions and helps in refactorings